cmd/copyEnumeratorHelper.go

```diff
@@ -84,46 +84,73 @@ var infoCopyFromBucketDirectoryListOfFiles = "trying to copy the source as bucke
 var infoCopyFromDirectoryListOfFiles = "trying to copy the source as directory/list of files"
 var infoCopyFromAccount = "trying to copy the source account"
 
-func handleSingleFileValidationErrorForBlob(err error) {
+func handleSingleFileValidationErrorForBlob(err error) (stop bool) {
 	errRootCause := err.Error()
 	stgErr, isStgErr := err.(azblob.StorageError)
 	if isStgErr {
-		if stgErr.ServiceCode() == azblob.ServiceCodeAuthenticationFailed {
+		if stgErr.ServiceCode() == azblob.ServiceCodeInvalidAuthenticationInfo {
+			errRootCause = fmt.Sprintf("%s, Please ensure all login details are correct (or that you are signed into the correct tenant for this blob)", stgErr.ServiceCode())
+			stop = true
+		} else if stgErr.ServiceCode() == azblob.ServiceCodeAuthenticationFailed {
 			errRootCause = fmt.Sprintf("%s, please check if SAS or OAuth is used properly, or source is a public blob", stgErr.ServiceCode())
+			stop = true
 		} else {
 			errRootCause = string(stgErr.ServiceCode())
 		}
 	}
 
-	glcm.Info(fmt.Sprintf(handleSingleFileValidationErrStr, errRootCause))
+	if stop {
```

Comment:
a bit confused here, why the different format depending on `stop`?

Reply:
So, the way that AzCopy checks if something is a file or folder is that it tries to error out when attempting to download a file first. If it errored out, it would give a strange error formatting in my opinion. I don't need to know that it isn't a file if I know that there's an authentication error, right?